packages/mask/src/components/shared/ApplicationBoard.tsx

```diff
@@ -24,54 +30,133 @@ const useStyles = makeStyles()((theme) => {
-                                cur.enableRequirement.networks.networks[currentSNSNetwork]
-                            const isSNSEnabled =
-                                currentSNSIsSupportedNetwork === undefined || currentSNSIsSupportedNetwork
+            <div className={classes.header}>
+                <Typography className={classes.subTitle}>{t('applications')}</Typography>
+                <img
+                    src={theme.palette.mode === 'dark' ? SettingIconDarkModeUrl : SettingIconLightModeUrl}
+                    className={classes.settingIcon}
+                    onClick={() => setOpenSettings(true)}
+                />
+            </div>
 
-                            return acc.concat(
-                                cur.ApplicationEntries.map((x) => {
-                                    return {
-                                        entry: x,
-                                        enabled: isSNSEnabled && (account ? isWeb3Enabled : !isWalletConnectedRequired),
-                                        pluginId: cur.ID,
-                                    }
-                                }) ?? [],
-                            )
-                        },
-                        [],
-                    )
-                    .sort((a, b) => a.entry.defaultSortingPriority - b.entry.defaultSortingPriority)
-                    .map(({ entry, enabled }, index) => (
-                        <Fragment key={index}>
-                            <entry.RenderEntryComponent disabled={!enabled} />
-                        </Fragment>
-                    ))}
-            </section>
+            {listedAppList.length > 0 ? (
+                <section className={classes.applicationWrapper}>
+                    {listedAppList.map((application) => {
+                        return (
+                            <Fragment key={application.pluginId}>
+                                <RenderEntryComponentWrapper application={application} />
+                            </Fragment>
+                        )
+                    })}
+                </section>
+            ) : (
+                <div className={classes.placeholderWrapper}>
+                    <Typography className={classes.placeholder}>
+                        {t('application_settings_tab_plug_app-unlisted-placeholder')}
+                    </Typography>
+                </div>
+            )}
+            {openSettings ? (
+                <ApplicationSettingDialog open={openSettings} onClose={() => setOpenSettings(false)} />
+            ) : null}
         </>
     )
 }
+
+interface RenderEntryComponentWrapperProps {
+    application: Application
+}
+
+function RenderEntryComponentWrapper({ application }: RenderEntryComponentWrapperProps) {
+    const RenderEntryComponent = application.entry.RenderEntryComponent!
+    return (
+        <RenderEntryComponent
+            disabled={!application.enabled}
+            icon={application.entry.icon}
```

Comment:
Why do you need to provide an icon back to itself? Can't the plugin just render the icon and title in the RenderEntryComponent they defined?

Reply:
done